nolearn/lasagne/tests/test_visualize.py

```diff
@@ -1,60 +1,78 @@
 import matplotlib.pyplot as plt
+import numpy as np
+import pytest
 
 
 class TestCNNVisualizeFunctions:
+    @pytest.fixture
+    def X_non_square(self, X_train):
+        X = np.hstack(
+            (X_train[:, :20 * 28], X_train[:, :20 * 28], X_train[:, :20 * 28]))
+        X = X.reshape(-1, 3, 20, 28)
+        return X
+
     def test_plot_loss(self, net_fitted):
         from nolearn.lasagne.visualize import plot_loss
         plot_loss(net_fitted)
         plt.clf()
         plt.cla()
 
-    def test_plot_conv_weights(self, net_fitted):
+    def plot_conv_weights(self, net, **kwargs):
         from nolearn.lasagne.visualize import plot_conv_weights
-        plot_conv_weights(net_fitted.layers_['conv1'])
-        plot_conv_weights(net_fitted.layers_['conv2'], figsize=(1, 2))
+        plot_conv_weights(net.layers_['conv1'], **kwargs)
         plt.clf()
         plt.cla()
 
-    def test_plot_conv_activity(self, net_fitted, X_train):
+    @pytest.mark.parametrize('kwargs', [{}, {'figsize': (3, 4)}])
+    def test_plot_conv_weights(self, net_fitted, net_color_non_square, kwargs):
+        # XXX workaround: fixtures cannot be used (yet) in conjunction
+        # with parametrize
```

Comment:
My initial intention was to pass one net at a time through an additional `parametrize`, like this:

```
@pytest.mark.parametrize('kwargs', [{}, {'figsize': (3, 4)}])
@pytest.mark.parametrize('net', [net_fitted, net_color_non_square])
def test_plot_conv_weights(self, net, ...)
```

i.e. using fixtures _in_ `parametrize`. It seems that this possibility is [considered](https://github.com/pytest-dev/pytest/issues/349); therefore, I put this comment there so that we can change it as soon as pytest allows this.